internal/app/frontend/frontend_test.go

```diff
@@ -32,3 +40,90 @@ func TestServerBinding(t *testing.T) {
 
 	rpcTesting.TestServerBinding(t, bs)
 }
+
+// validateTicket validates that the fetched ticket is identical to the expected ticket.
+func validateTicket(t *testing.T, got *pb.Ticket, want *pb.Ticket) {
+	assert.Equal(t, got.Id, want.Id)
+	assert.Equal(t, got.Properties, want.Properties)
+	assert.Equal(t, got.Assignment.Connection, want.Assignment.Connection)
+	assert.Equal(t, got.Assignment.Properties, want.Assignment.Properties)
+	assert.Equal(t, got.Assignment.Error, want.Assignment.Error)
+}
+
+// validateDelete validates that the ticket is actually deleted from the state storage.
+// Given that delete is async, this method retries fetch every 100ms up to 5 seconds.
+func validateDelete(t *testing.T, fe *frontendService, id string) {
+	start := time.Now()
+	for {
+		if time.Since(start) > 5*time.Second {
+			break
+		}
+
+		// Attempt to fetch the ticket every 100ms
+		_, err := fe.GetTicket(context.Background(), &pb.GetTicketRequest{TicketId: id})
+		if err != nil {
+			// Only failure to fetch with NotFound should be considered as success.
+			assert.Equal(t, status.Code(err), codes.NotFound)
+			return
+		}
+
+		time.Sleep(100 * time.Millisecond)
+	}
+
+	assert.Failf(t, "ticket %v not deleted after 5 seconds", id)
+}
+
+// TestFrontendService tests creating, getting and deleting a ticket using Frontend service.
+func TestFrontendService(t *testing.T) {
```

Comment:
This looks more like a statestore test. Is it possible to use a frontend client for testing?

Reply:
What do you mean? This test fetches a frontend client from newFrontend and then tests all methods only from the frontend client. The frontend client will always need a state store - so yes, we are creating one here - but this does test FE Client methods. We should have E2E tests that talk via APIs to the frontend service - but this is just testing the client. (I am hoping our integration tests account for those)